plugins/generic/pln/PLNGatewayPlugin.inc.php

```diff
@@ -111,28 +112,52 @@ function getManagementVerbs() {
 	 * Handle fetch requests for this plugin.
 	 */
 	function fetch() {
-                $templateMgr =& TemplateManager::getManager();
+		$plugin = & $this->getPLNPlugin();
+		$templateMgr = & TemplateManager::getManager();
+
+		$journal = & Request::getJournal();
+		$templateMgr->assign_by_ref('journal', $journal);
+
+		$pluginVersionFile = $this->getPluginPath() . DIRECTORY_SEPARATOR . '/version.xml';
+		$pluginVersion = & VersionCheck::parseVersionXml($pluginVersionFile);
+		$templateMgr->assign_by_ref('pluginVersion', $pluginVersion);
+
+		$terms = array();
+		$termsAccepted = $plugin->termsAgreed($journal->getId());
+		if ($termsAccepted) {
+			$templateMgr->assign('termsAccepted', 'yes');
+			$terms = unserialize($plugin->getSetting($journal->getId(), 'terms_of_use'));
+			$termsAgreement = unserialize($plugin->getSetting($journal->getId(), 'terms_of_use_agreement'));
+		} else {
+			$templateMgr->assign('termsAccepted', 'no');
+		}
+
+		$termKeys = array_keys($terms);
+		$termsDisplay = array();
+		foreach ($termKeys as $key) {
+			$termsDisplay[] = array(
+				'key' => $key,
+				'term' => $terms[$key]['term'],
+				'updated' => $terms[$key]['updated'],
+				'accepted' => $termsAgreement[$key]
+			);
+		}
+		$templateMgr->assign_by_ref('termsDisplay', new ArrayItemIterator($termsDisplay));
```

Comment:
I don't know if the results of the "new" operator could be assigned by reference in php4. Safer to just use `->assign`, I think.